### PR TITLE
refactor: code refactoring in HealthCheck.php

### PR DESCRIPTION
**Pull request by DiegoHBustamante**
This version includes the following change:

- Code refactoring HealthCheck.php.
- Delete unused code.
- Code simplificaction.
Captura de pantalla 2024-08-27 a la(s) 11 28 53 a  m

### CODE REVIEW

File: src/transbank_webpay_rest/library/HealthCheck.php [hunk 1]
```diff
@@ -26,61 +26,62 @@ public function __construct($config)
...
-            $this->versioninfo = [
-                'status'  => 'Error!: Versión no soportada',
-                'version' => phpversion(),
-            ];
-        }
+        $minVersion = '7.0.0';
+        $maxVersion = '7.4.0';
+        $currentVersion = phpversion();
+        $isValidVersion = version_compare($currentVersion, $minVersion, '>=') && version_compare($currentVersion, $maxVersion, '<=');
+        $this->versioninfo = [
```
Comment (Matiasnickolas) [inline src/transbank_webpay_rest/library/HealthCheck.php]: is necessary set this `versionInfo` as an attribute? or can return this directly

File: src/transbank_webpay_rest/library/HealthCheck.php [hunk 2]
```diff
@@ -26,61 +26,62 @@ public function __construct($config)
...
+        return [
+            'status' => 'OK',
+            'version' => $version
+        ];
     }
 
-    // obtiene ultima version exclusivamente para virtuemart
-    // NOTE: lastrelasevirtuemart
+    /**
+     * Gets the currently installed Virtuemart version.
```
Comment (Matiasnickolas) [inline src/transbank_webpay_rest/library/HealthCheck.php]: this returns last virtuemart version, not the installed version

File: src/transbank_webpay_rest/library/HealthCheck.php [hunk 3]
```diff
@@ -101,26 +102,32 @@ private function getLastVirtuemartVersion()
...
             exit;
-        } else {
-            $xml = simplexml_load_file(JPATH_PLUGINS . '/vmpayment/transbank_webpay_rest/transbank_webpay_rest.xml', null, LIBXML_NOCDATA);
-            $json = json_encode($xml);
-            $arr = json_decode($json, true);
-            $currentplugin = $arr['version'];
         }
+        $xml = simplexml_load_file($pluginPath, null, LIBXML_NOCDATA);
+        if ($xml === false) {
+            exit;
```
Comment (Matiasnickolas) [inline src/transbank_webpay_rest/library/HealthCheck.php]: is safe use exit instead return?

File: src/transbank_webpay_rest/library/HealthCheck.php [hunk 4]
```diff
@@ -26,61 +26,62 @@ public function __construct($config)
...
         ];
     }
 
-    // valida version de php
+    /**
+     * Validates the current PHP version.
+     *
+     * @return array The status and the current PHP version.
+     */
     private function getValidatephp()
```
Comment (mastudillot) [inline src/transbank_webpay_rest/library/HealthCheck.php]: Use a descriptive name, is a get or a validate function?

**Review by mastudillot — CHANGES_REQUESTED** [review]
add types in variable and functions

**Review by mvarlic — APPROVED** [review]
LGTM!

File: src/transbank_webpay_rest/library/HealthCheck.php [hunk 5]
```diff
@@ -144,55 +152,76 @@ private function getLastGitHubReleaseVersion($repository): string
...
+    private function getPluginInfo($ecommerce): array
     {
-        $data = $this->getEcommerceInfo($ecommerce);
+        $ecommerceInfo = $this->getEcommerceInfo();
         return [
             'ecommerce'              => $ecommerce,
-            'ecommerce_version'      => $data['current_ecommerce_version'],
-            'current_plugin_version' => $data['current_plugin_version'],
+            'ecommerce_version'      => $ecommerceInfo ? $ecommerceInfo['current_ecommerce_version'] : 'ERROR: No se pudo recuperar la versión de Ecommerce',
+            'current_plugin_version' => $ecommerceInfo ? $ecommerceInfo['current_plugin_version'] : 'ERROR: No se pudo recuperar la versión de Ecommerce',
```
Comment (Matiasnickolas) [inline src/transbank_webpay_rest/library/HealthCheck.php]: This error message is related to ecommerce version or tbk plugin version?

**Review by mastudillot — APPROVED** [review]
LGTM

**Pull request merged**